xServerWorker/BackgroundServices/BlockProcessingWorker.cs

```diff
@@ -0,0 +1,491 @@
+
+
+#else
+            _client = new MongoClient($"mongodb://{mongoUser}:{mongoPassword}@xDocumentStore:27017/");
+            //_xServerHost = Environment.GetEnvironmentVariable("XSERVER_BACKEND") ?? "";
+
+#endif
+
+
+
+            _db = _client.GetDatabase("xServerDb");
+        }
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            try
+            {
+                var blockCount = await GetBlockCount();
+                _feeAddress = await GetMyFeeAddress();
+
+                var dictionaryCollection = _db.GetCollection<BsonDocument>("Dictionary");
+
+
+                var filter = Builders<BsonDocument>.Filter.Eq("_id", "latestBlock");
+                var document = dictionaryCollection.Find(filter).FirstOrDefault();
+
+                if (document != null)
+                {
+                    var id = document["value"].ToString();
+
+                    if (id != null)
+                    {
+                        latestBlockProcessed = int.Parse(id);
+
+                    }
+
+                }
+                else
+                {
+                    dynamic lastBlockEntry = new ExpandoObject();
+                    lastBlockEntry._id = "latestBlock";
+                    lastBlockEntry.value = latestBlockProcessed;
+
+
+                    var jsonstring = JsonConvert.SerializeObject(lastBlockEntry);
+                    var updateDocument = BsonSerializer.Deserialize<BsonDocument>(jsonstring);
+
+                    dictionaryCollection.InsertOne(updateDocument);
+                    document = updateDocument;
+
+                }
+
+
+                _logger.LogInformation($"Latest block processed : {latestBlockProcessed}");
+                _logger.LogInformation($"Latest block height : {blockCount}");
+
+                var taskList = new List<Task<string>>();
+                var consinueTaskList = new List<Task>();
+
+
+                _logger.LogInformation("Processsing Blocks...");
+
+
+
+
+                while (!stoppingToken.IsCancellationRequested)
+                {
+                    if (latestBlockProcessed < blockCount && _feeAddress != "")
+                    {
+                        await ProcessBlock(stoppingToken);
+
+                        if (document != null)
+                        {
+                            var builder = Builders<BsonDocument>.Filter;
+
+                            filter = builder.Eq("_id", "latestBlock");
+                            var update = Builders<BsonDocument>.Update.Set("value", latestBlockProcessed);
+                            dictionaryCollection.UpdateOne(filter, update);
+
+                        }
+
+                    }
+                    else
+                    {
+
+
+                        blockCount = await GetBlockCount();
+
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+
+                throw;
```

Comment:
Catch Throw will lose the exception stack.
Better to remove the block, or if you will keep, log it.